numpy/core/src/multiarray/item_selection.c

```diff
@@ -2124,6 +2129,46 @@ count_nonzero_bytes_384(const npy_uint64 * w)
     return r;
 }
 
+#if NPY_SIMD
+static NPY_INLINE npy_intp
+count_nonzero_bytes(const npy_uint8 *d, npy_uintp unrollx)
+{
+    int count = 0;
+    int i = 0;
+    
+    const int vstep = npyv_nlanes_u8;
+    const npyv_u8 vone = npyv_setall_u8(1);
+    const npyv_u8 vzero = npyv_setall_u8(0);
+    npyv_u8 vt;
+    npyv_u32 vsum32 = npyv_zero_u32();
+    while (i < unrollx)
+    {
+        npyv_u16 vsum16 = npyv_zero_u16();
+        int j = i;
+        while (j < MIN(unrollx, i + 65280 * npyv_nlanes_u16))
+        {
+            int k = j;
+            npyv_u8 vsum8 = npyv_zero_u8();
+            for (; k < MIN(unrollx, j + 255 * vstep); k += vstep)
```

Comment:
Let's make the intent here a little clearer:

```suggestion
        while (j < MIN(unrollx, i + 0xFF00 * npyv_nlanes_u16))
        {
            int k = j;
            npyv_u8 vsum8 = npyv_zero_u8();
            for (; k < MIN(unrollx, j + 0xFF * vstep); k += vstep)
```